modules/nlu/src/backend/tools/token-utils.ts

```diff
@@ -0,0 +1,76 @@
+import _ from 'lodash'
+import { Token } from '../typings'
+
+const SPACE = '\u2581'
+
+export const makeTokens = (stringTokens: string[], text: string) => {
+  return stringTokens.reduce(reduceTokens(text), [] as Token[])
+}
+
+const reduceTokens = (text: string) => (currentTokens: Token[], token: string) => {
+  const trimedToken = token.replace(SPACE, '')
+
+  const previousToken = currentTokens[currentTokens.length - 1]
+  const cursor = previousToken ? previousToken.end : 0
+
+  const cutText = text.substring(cursor).toLowerCase()
+  const start = cutText.indexOf(trimedToken) + cursor
+  const sanitized = text.substr(start, trimedToken.length)
+
+  const newToken = {
+    value: token,
+    cannonical: sanitized,
+    start,
+    end: start + trimedToken.length,
+    matchedEntities: []
+  } as Token
+
+  return currentTokens.concat(newToken)
+}
+
+export const mergeSpecialCharactersTokens = (toks: Token[], specialCharacters: string[]) => {
+  const isMergeable = ({ value: previous }: Token, { value: current }: Token) => {
+    previous = previous.replace(SPACE, '')
+    return [previous, current].reduce((ac: boolean, tok: string) => {
+      return ac && tokenIsAllMadeOf(tok, specialCharacters)
+    }, true)
+  }
+
+  return mergeTokens(toks, isMergeable)
+}
```

Comment:
The only thing I would change is declare isMergeable outside mergeSpecialCharactersTokens.
Such as;
```js

const isMergeable = (specialCharacters) => ({ value: previous }: Token, { value: current }: Token) => {
    previous = previous.replace(SPACE, '')
    return [previous, current].reduce((ac: boolean, tok: string) => {
      return ac && tokenIsAllMadeOf(tok, specialCharacters)
    }, true)
  }

export const mergeSpecialCharactersTokens = (toks: Token[], specialCharacters: string[]) => mergeTokens(toks, isMergeable(specialCharacters))
```